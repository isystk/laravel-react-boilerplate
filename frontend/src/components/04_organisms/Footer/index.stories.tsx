import { Meta, Story } from '@storybook/react'
import React from 'react'
import Footer, { FooterProps } from './index'

export default {
  title: '04_organisms/Footer',
  component: Footer,
} as Meta

const Template: Story = () => {
  const props: FooterProps = {
    menuItems: [
      {
        label: 'Menu1',
        href: 'https://menu1',
      },
      {
        label: 'Menu2',
        href: 'https://menu2',
      },
      {
        label: 'Menu3',
        href: 'https://menu3',
        target: '_blank',
      },
    ],
  }
  return <Footer {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
