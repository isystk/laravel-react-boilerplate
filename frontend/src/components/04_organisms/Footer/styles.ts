import { css, keyframes } from '@emotion/css'

export const footer = css``
