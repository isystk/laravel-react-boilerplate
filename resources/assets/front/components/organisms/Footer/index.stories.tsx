import Footer from './index';
import { JSX } from 'react';

export default {
  title: 'Components/Organisms/Footer',
  component: Footer,
  tags: ['autodocs'],
};

export const Default: { render: () => null | JSX.Element } = {
  render: () => <Footer />,
};
