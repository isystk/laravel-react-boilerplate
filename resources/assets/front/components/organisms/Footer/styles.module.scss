@use "@/assets/styles/variable" as *;

.footer {
    background-color: $dark-gray;
    padding: 40px 0
}

.footerInner {
    text-align: center;
    img {
        margin: auto;
    }
}

.footerCopy {
    font-size: 0.8rem;
    margin: 20px;
}

@media only screen and (min-width: 767px) {
    .footer {
        padding: 60px 0
    }
    .footerCopy {
        font-size: 1rem
    }
}
