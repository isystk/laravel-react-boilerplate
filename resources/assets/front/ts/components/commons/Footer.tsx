import * as React from "react";
import { FC } from "react";

const CommonFooter: FC = () => {
    return (
        <footer className="footer">
            <div className="footer_inner">
                <p className="mt20">
                    <img src="/assets/front/image/logo_02.png" alt="" />
                </p>
                <p className="mt10">
                    <small className="fz-s">
                        ©️isystk
                        このページは架空のページです。実際の人物・団体とは関係ありません。
                    </small>
                </p>
            </div>
        </footer>
    );
};

export default CommonFooter;
