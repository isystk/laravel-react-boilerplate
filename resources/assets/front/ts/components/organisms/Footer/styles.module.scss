@use "@/styles/variable";

.footer {
    background-color: #dcdcdc;
    padding: 40px 0
}

.footerInner {
    text-align: center;
    img {
        margin: auto;
    }
}

.footerCopy {
    font-size: 0.8rem;
    margin: 20px;
}

@media only screen and (min-width: 767px) {
    .footer {
        padding: 60px 0
    }
    .footerCopy {
        font-size: 1rem
    }
}
