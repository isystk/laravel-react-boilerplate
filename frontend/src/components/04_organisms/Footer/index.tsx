import React, { FC } from 'react'
import { ContainerProps } from 'types'
import * as styles from './styles'
import { connect } from '@/components/hoc'
import LineButton from '@/components/01_atoms/LineButton'
import ScrollTop from '@/components/01_atoms/ScrollTop'
import Image from '@/components/01_atoms/Image'
import { useI18n } from '@/components/i18n'
import { type MenuItem } from '@/constants/menu'

/** FooterProps Props */
export type FooterProps = { menuItems: MenuItem[] }
/** Presenter Props */
export type PresenterProps = FooterProps & {
  t
}

/** Presenter Component */
const FooterPresenter: FC<PresenterProps> = ({
  t,
  menuItems = [],
  ...props
}) => (
  <>
    <footer className={`${styles.footer} relative`}>
      <div className="absolute left-0 right-0 -mt-64 md:-mt-32">
        <div className="flex w-full md:w-4/5 bg-main rounded-tr-full py-24 px-8 pb-16"></div>
        <div className="flex flex-wrap w-full md:w-4/5 bg-main md:px-40 pb-16">
          <div className="flex w-full md:w-1/2 justify-center md:justify-end items-center md:pr-8 mb-8">
            <Image
              src="https://qr-official.line.me/gs/M_618ecvqb_GW.png"
              alt="line-me"
            />
          </div>
          <div className="flex w-full md:w-1/2 justify-center md:justify-start items-center md:pl-8">
            <div className="px-8 md:px-0">
              <p className="text-accent text-xl md:text-3xl font-bold mb-6 ">
                {t(
                  'Why is the sky blue? Tell me a delicious recipe! and more!'
                )}
              </p>
              <LineButton
                link="#"
                label={t('Add me as a friend and ask me a question')}
              />
            </div>
          </div>
        </div>
      </div>
      <div className="bg-accent rounded-tl-full py-36 px-8 pb-16"></div>
      <div className="bg-accent py-96 px-8 pb-16">
        <nav className="flex flex-wrap items-center justify-center">
          {menuItems.map(({ label, href, target }, idx) => (
            <a
              href={href}
              target={target}
              rel={target ? 'noreferrer' : ''}
              className="w-full mb-6 md:w-48"
              key={idx}
            >
              <p className="text-white text-center font-bold">
                {'Twitter' === label ? (
                  <svg
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    className="text-white h-6 w-6 m-auto"
                  >
                    <path d="M8.12 20.329c7.75 0 11.99-6.414 11.99-11.989v-.546c.823-.597 1.534-1.336 2.097-2.184-.759.338-1.562.566-2.396.668a4.192 4.192 0 0 0 1.836-2.315 8.331 8.331 0 0 1-2.65 1.011 4.156 4.156 0 0 0-7.113 3.795 11.817 11.817 0 0 1-8.583-4.34 4.156 4.156 0 0 0 1.288 5.559A4.103 4.103 0 0 1 2.3 7.627v.055a4.156 4.156 0 0 0 3.32 4.072 4.16 4.16 0 0 1-1.868.071 4.155 4.155 0 0 0 3.87 2.878 8.343 8.343 0 0 1-5.164 1.78A8.756 8.756 0 0 1 0 16.736 11.722 11.722 0 0 0 8.12 20.33"></path>
                  </svg>
                ) : (
                  t(label)
                )}
              </p>
            </a>
          ))}
        </nav>
        <div className="flex items-center justify-center ">
          <p className="text-white text-center text-sm">
            Â© 2023 Ise Inc. All Right Reserved. Illustration by Yoshitaka Ise
          </p>
        </div>
      </div>
    </footer>
    <ScrollTop scrollY={2000} />
  </>
)

/** Container Component */
const FooterContainer: React.FC<
  ContainerProps<FooterProps, PresenterProps>
> = ({ presenter, ...props }) => {
  const { t } = useI18n('Common')

  return presenter({
    t,
    ...props,
  })
}

export default connect<FooterProps, PresenterProps>(
  'Footer',
  FooterPresenter,
  FooterContainer
)
