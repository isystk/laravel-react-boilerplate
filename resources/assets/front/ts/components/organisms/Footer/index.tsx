import styles from './styles.module.scss';

const Footer = () => {
    return (
        <footer className={styles.footer}>
            <p>&copy; 2025 ISE. All Rights Reserved</p>
        </footer>
    );
};

export default Footer;

