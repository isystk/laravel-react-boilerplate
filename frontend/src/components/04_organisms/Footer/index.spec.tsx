import React from 'react'
import renderer from 'react-test-renderer'
import Footer, { FooterProps } from './index'

describe('Footer', () => {
  it('Match Snapshot', () => {
    const props: FooterProps = {
      menuItems: [
        {
          label: 'Menu1',
          href: 'https://menu1',
        },
        {
          label: 'Menu2',
          href: 'https://menu2',
        },
        {
          label: 'Menu3',
          href: 'https://menu3',
          target: '_blank',
        },
      ],
    }
    const component = renderer.create(<Footer {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
