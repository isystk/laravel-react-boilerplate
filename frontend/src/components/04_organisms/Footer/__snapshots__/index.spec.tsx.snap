// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Match Snapshot 1`] = `
[
  <footer
    className="css-0 relative"
  >
    <div
      className="absolute left-0 right-0 -mt-64 md:-mt-32"
    >
      <div
        className="flex w-full md:w-4/5 bg-main rounded-tr-full py-24 px-8 pb-16"
      />
      <div
        className="flex flex-wrap w-full md:w-4/5 bg-main md:px-40 pb-16"
      >
        <div
          className="flex w-full md:w-1/2 justify-center md:justify-end items-center md:pr-8 mb-8"
        >
          <div>
            <img
              alt="line-me"
              className="css-s08p0c "
              loading="lazy"
              src="https://qr-official.line.me/gs/M_618ecvqb_GW.png"
            />
          </div>
        </div>
        <div
          className="flex w-full md:w-1/2 justify-center md:justify-start items-center md:pl-8"
        >
          <div
            className="px-8 md:px-0"
          >
            <p
              className="text-accent text-xl md:text-3xl font-bold mb-6 "
            >
              Why is the sky blue? 
              <br />
               Tell me a delicious recipe! 
              <br />
               and more!
            </p>
            <a
              className="css-0 p-2 pl-4 pr-4 flex items-center justify-center bg-accent rounded-full"
              href="#"
            >
              <img
                alt="Add me as a friend and ask me a question"
                height="50px"
                src="/images/icon-line.png"
                width="50px"
              />
              <span
                className="text-white font-bold"
              >
                Add me as a friend and ask me a question
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bg-accent rounded-tl-full py-36 px-8 pb-16"
    />
    <div
      className="bg-accent py-96 px-8 pb-16"
    >
      <nav
        className="flex flex-wrap items-center justify-center"
      >
        <a
          className="w-full mb-6 md:w-48"
          href="https://menu1"
          rel=""
        >
          <p
            className="text-white text-center font-bold"
          >
            Menu1
          </p>
        </a>
        <a
          className="w-full mb-6 md:w-48"
          href="https://menu2"
          rel=""
        >
          <p
            className="text-white text-center font-bold"
          >
            Menu2
          </p>
        </a>
        <a
          className="w-full mb-6 md:w-48"
          href="https://menu3"
          rel="noreferrer"
          target="_blank"
        >
          <p
            className="text-white text-center font-bold"
          >
            Menu3
          </p>
        </a>
      </nav>
      <div
        className="flex items-center justify-center "
      >
        <p
          className="text-white text-center text-sm"
        >
          Â© 2023 Ise Inc. All Right Reserved. Illustration by Yoshitaka Ise
        </p>
      </div>
    </div>
  </footer>,
  <button
    className="css-0 fixed bottom-6 md:bottom-20 right-6 md:right-20 w-12 md:w-16 h-12 md:h-16 border-none bg-main rounded-full cursor-pointer transition-opacity duration-200 ease-in-out css-14nb29t"
    onClick={[Function]}
  >
    <span
      className="text-bold text-gray-600"
    >
      ^
    </span>
  </button>,
]
`;
